templates/problem/voting-controls.html

```diff
@@ -0,0 +1,218 @@
+            $.ajax({
+                url: '{{ url('delete_vote', object.code) }}',
+                type: 'POST',
+                data: $('#id_delete_vote_form').serialize(),
+                success: () => {
+                    deleteUserVote(voted_points);
+                    voted_points = null;
+                    voted_note = null;
+                    deleteVoteUpdate();
+                    // Forms are auto disabled to prevent resubmission, but we need to allow resubmission here.
+                    $('id_delete_vote_form_submit_button').removeAttr('disabled');
+                    closeLightbox('id_voting_lightbox');
+                },
+                error: data => {
+                    alert('Unable to delete vote: '+data.responseText);
+                },
+            });
+        });
+
+        $('#id_vote_form').on('submit', e => e.preventDefault());
+        $('#id_vote_form_submit_button').on('click', e => {
+            e.preventDefault();
+            $.ajax({
+                url: '{{ url('vote', object.code) }}',
+                type: 'POST',
+                data: $('#id_vote_form').serialize(),
+                success: data => {
+                    updateUserVote(voted_points, data.points);
+                    voted_note = data.note;
+                    voted_points = data.points;
+                    voteUpdate();
+                    // Forms are auto disabled to prevent resubmission, but we need to allow resubmission here.
+                    $('#id_vote_form_submit_button').removeAttr('disabled');
+                    closeLightbox('id_voting_lightbox');
+                },
+                error: data => {
+                    let errors = data.responseJSON;
+                    if('points' in errors){
+                        $('#id_points_error_box').show();
+                        $('#id_points_error').prop('textContent', errors.points[0]);
+                    } else {
+                        $('#id_points_error_box').hide();
+                    }
+                    if('note' in errors){
+                        $('#id_note_error_box').show();
+                        $('#id_note_error').prop('textContent', errors.note[0]);
+                    } else {
+                        $('#id_note_error_box').hide();
+                    }
+                }
+            });
+        });
+
+        createLightbox('id_voting_lightbox', 'id_vote_form_box');
+        $('#id_vote_button').on('click', () => activateLightbox('id_voting_lightbox'));
+    </script>
+{% endif %}
+
+{% if not request.in_contest %}
+    {% include 'problem/voting-stats.html' %}
+    <a class="form-button" id="id_vote_stats_button">{{ _('Voting statistics') }}</a>
+    <script>
+        createLightbox('id_vote_stats_lightbox', 'id_vote_stats')
+        $('#id_vote_stats_button').on('click', () => {
+            activateLightbox('id_vote_stats_lightbox');
+            reload_vote_graph();
+        });
+    </script>
+{% endif %}
```

Comment:
There's no way this MASSIVE JS code block deserves to lie in an HTML file.

Reply:
Looking at the dmoj code-base, it seems they put their js in a file called something-js.html instead of a .js file, should I also do that instead of putting in a .js file?